packages/php-wasm/web/src/lib/register-service-worker.ts

```diff
@@ -45,6 +54,13 @@ export async function registerServiceWorker<Client extends Remote<PHPWorker>>(
 	// the update:
 	await registration.update();
 
+	registration.addEventListener('worker-api', (event: CustomEventInit) => {
+		if (!event.detail.workerApi) {
+			return;
+		}
+		phpApi = event.detail.workerApi;
+	});
+
 	// Proxy the service worker messages to the web worker:
 	navigator.serviceWorker.addEventListener(
```

Comment:
This will throw errors until `phpApi` is available. Let's make it await a promise that resolves once `phpApi` is available. Perhaps the `registerServiceWorker` could accept that promise as an argument.

Reply:
In my testing, it never failed, but just to make sure, I added an error if it happens and a promise to await https://github.com/WordPress/wordpress-playground/pull/1606/commits/b167feb485a648e171dc79078528908f74b7a050

Reply:
I've made a few adjustments in [`c3ca40f` (#1606)](https://github.com/WordPress/wordpress-playground/pull/1606/commits/c3ca40fdbc9544b3fa62eafa08323281de125ece)